### PR TITLE
Remove dark mode flicker in (almost) all situations

### LINKED ISSUE #74: There is a flicker of light -> dark theme on page load
I imagine this is due to the following steps taking place:

1. Render in light theme on the server
2. Hydrate in client
3. Pickup on user preference, switch to dark mode

No idea of what the solution is for this but it's quite jarring on slower connections as it stays in light theme while the JS is downloaded.

### PR DESCRIPTION
**Pull request by tom-sherman**
Fixes #74

This fix solves this problem in all situations unless:

- You have selected light mode but your OS is set to dark mode (or vice versa) - maybe this could be solved by storing the preference in a cookie and SSRing

or

- Your browser does not support the `prefers-color-scheme` media query and you have toggled to dark mode in a previous session/tab - this could be solved by not rendering on the server in these browsers I guess 😬

### CODE REVIEW

File: src/css/tailwind.css [hunk 1]
```diff
@@ -47,6 +47,42 @@ path.logo-nor {
   fill: theme("colors.gray.800");
 }
 
+@media (prefers-color-scheme: dark) {
```
Comment (tom-sherman) [inline src/css/tailwind.css:50]: These styles are duplicated from above which isn't ideal, if you have any ideas about how they could be kept in sync I'd love to hear them! I'm not familiar with the CSS setup you're running here

File: src/components/DarkModeToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import useDarkMode from "use-dark-mode";
+
+/**
+ * This component should only be called on the client as it accesses window in it's body
+ */
+export default function DarkModeToggle() {
+  const darkMode = useDarkMode(
+    // @ts-ignore use-dark-mode is passing this value directly to useState, so the types are actually wrong here it should accept a lazy initial state
```
Comment (tom-sherman) [inline src/components/DarkModeToggle.tsx:8]: I'm gonna submit a PR to `use-dark-mode` to update the types, hopefully they accept it.

**Comment by tom-sherman** [comment]
Oh! I also haven't tested switching OS color schemes, and I've read some bugs relating to this over at `use-dark-mode`. I feel like this is too much of an edge case to care about 😆

**Review by Illizian — APPROVED** [review]

File: src/components/DarkModeToggle.tsx (hunk 2 above)
Comment (tom-sherman) [reply src/components/DarkModeToggle.tsx:8]: For prosperity: https://github.com/donavon/use-dark-mode/pull/75

**Pull request merged**